app/code/community/Bolt/Boltpay/Model/ShippingAndTax.php

```diff
@@ -43,11 +43,16 @@ public function applyShippingAddressToQuote( $quote, $shippingAddress ) {
             throw new Exception(Mage::helper('boltpay')->__("Address must contain postal_code and country_code."));
         }
 
+        $shippingStreet = (@$shippingAddress->street_address1 ?: '') . "\n"
```

Comment:
can you do something like:

```
implode('\n', array_filter(array(
  @$shippingAddress->street_address1,
  @$shippingAddress->street_address2,
  @$shippingAddress->street_address3,
  @$shippingAddress->street_address4))
```

Your test results in `'Sample Street 10' . "\n" . 'Apt 123'` instead of `'Sample Street 10' . "\n" . 'Apt 123\n\n'`

Reply:
ok fair point. how about `trim` then? magento does trim.
https://github.com/mgtf/magento-1.9.2.0/blob/b5c1fc794d477f7d59b43c8740da0c9597e047a8/app/code/core/Mage/Customer/Model/Address/Abstract.php#L162

will it cause issue if newline code at the end is missing?

It seems weird to me that we add extra new lines code but if we need these for make magento happy I'm ok

Reply:
@daisy1754 trim will work.

Reply:
can we trim in plugin, not in test?

I want to understand if we need trailing newlines for `Sample Street 10\nApt 123\n\n`. If not needed, I'd prefer removing these. It may cause weird UI (extra spaces) somewhere.

Reply:
> can we trim in plugin, not in test?
> 
> I want to understand if we need trailing newlines for `Sample Street 10\nApt 123\n\n`. If not needed, I'd prefer removing these. It may cause weird UI (extra spaces) somewhere.

Ok.  I will ignore this:

>Additionally, some may have active validation on the values of street address line 3 and street address line 4, checking if it is a string which is likely done through the event customer_address_validation_after.
>Here, as a JIC (just-in-case), we provide an empty string value for these fields as our default.

Instead, if an error happens in validation, we can consider it as a need for merchant specific custom code.